docs/storage.md

```diff
@@ -93,6 +92,11 @@ config:
   trace:
     enable: false
   part_size: 134217728
+  sse_config:
+    enabled: false
```

Comment:
This looks interesting. I assume we cannot just remove this field and enable when any of below got specified as basic SSE does not need any options. The problem is that the validation and behaviour for this might be quite non trivial. For example what if enabled is false and enc key is set? (: 


I would propose some enum here:
```
sse_config:
    enabled: false
 

 
GIF
▲
This looks interesting. I assume we cannot just remove this field and enable when any of below got specified as basic SSE does not need any options. The problem is that the validation and behaviour for this might be quite non trivial. For example what if enabled is false and enc key is set? (: 


I would propose some enum here:
```
    type: "SEE-S3|SSE-KMS|SSE-C"
    kms_key_id: ""
    kms_encryption_context: {}
    encryption_key: ""
```
Might be more verbose and the behavior might be more clear. E.g if KMS is specified we ignore other options (or throw error if others specified). WDYT?

Reply:
I agree; right now the logic is implicit and requires knowledge of how the feature works. Changing the SSE to an explicit type will make the logic/validation clearer (you get what you want, we ignore the rest).

Will implement tomorrow.